forgerock-openbanking-uk-aspsp-rs/forgerock-openbanking-uk-aspsp-rs-gateway/forgerock-openbanking-uk-aspsp-rs-gateway-server/src/main/java/com/forgerock/openbanking/aspsp/rs/api/payment/v3_1_4/domesticpayments/DomesticPaymentConsentsApi.java

```diff
@@ -28,15 +28,179 @@
-import org.springframework.web.bind.annotation.RequestMapping;
+import com.forgerock.openbanking.exceptions.OBErrorResponseException;
+import io.swagger.annotations.*;
+import org.joda.time.DateTime;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+import uk.org.openbanking.datamodel.error.OBErrorResponse1;
+import uk.org.openbanking.datamodel.payment.OBWriteDomesticConsent4;
+import uk.org.openbanking.datamodel.payment.OBWriteDomesticConsentResponse4;
+import uk.org.openbanking.datamodel.payment.OBWriteFundsConfirmationResponse1;
 
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.Valid;
+import java.security.Principal;
+
+import static com.forgerock.openbanking.constants.OpenBankingConstants.HTTP_DATE_FORMAT;
+
+@javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.SpringCodegen", date = "2020-12-19T16:47:08.987291Z[Europe/London]")
 @Api(value = "domestic-payment-consents", description = "the domestic-payment-consents API")
 @OpenBankingAPI(
         obVersion = "3.1.4",
         obGroupName = OBGroupName.PISP,
         obReference = OBReference.DOMESTIC_PAYMENTS
 )
 @RequestMapping(value = "/open-banking/v3.1.4/pisp")
-public interface DomesticPaymentConsentsApi extends com.forgerock.openbanking.aspsp.rs.api.payment.v3_1_3.domesticpayments.DomesticPaymentConsentsApi {
-}
\ No newline at end of file
+public interface DomesticPaymentConsentsApi {
+
+    @ApiOperation(value = "Create Domestic Payment Consents", nickname = "createDomesticPaymentConsents", notes = "", response = OBWriteDomesticConsentResponse4.class, authorizations = {
+            @Authorization(value = "TPPOAuth2Security", scopes = {
+                    @AuthorizationScope(scope = "payments", description = "Generic payment scope")
+            })
+    }, tags = {"Domestic Payments",})
+    @ApiResponses(value = {
+            @ApiResponse(code = 201, message = "Domestic Payment Consents Created", response = OBWriteDomesticConsentResponse4.class),
+            @ApiResponse(code = 400, message = "Bad request", response = OBErrorResponse1.class),
+            @ApiResponse(code = 401, message = "Unauthorized"),
+            @ApiResponse(code = 403, message = "Forbidden", response = OBErrorResponse1.class),
+            @ApiResponse(code = 405, message = "Method Not Allowed"),
+            @ApiResponse(code = 406, message = "Not Acceptable"),
+            @ApiResponse(code = 415, message = "Unsupported Media Type"),
+            @ApiResponse(code = 429, message = "Too Many Requests"),
+            @ApiResponse(code = 500, message = "Internal Server Error", response = OBErrorResponse1.class)})
+    @PreAuthorize("hasAuthority('ROLE_PISP')")
+    @OpenBankingAPI(
+            obReference = OBReference.CREATE_DOMESTIC_PAYMENT_CONSENT
+    )
```

Comment:
Looks good. There are lots of little intricate things to get right and this all seems fine to me 👍🏻